nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java

```diff
@@ -3227,4 +3227,36 @@ public void testCustomNonnullAnnotation() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void testMapGetChainWithCast() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Constants.java",
+            "package com.uber;",
+            "public class Constants {",
+            "   public static final String KEY_1 = \"key1\";",
+            "   public static final String KEY_2 = \"key2\";",
+            "   public static final String KEY_3 = \"key3\";",
+            "}")
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import java.util.Map;",
+            "class Test {",
+            "   boolean withoutCast(Map<String, Map<String, Map<String, Object>>> topLevelMap){",
+            "     return topLevelMap.get(Constants.KEY_1) == null ",
+            "       || topLevelMap.get(Constants.KEY_1).get(Constants.KEY_2) == null",
+            "       || topLevelMap.get(Constants.KEY_1).get(Constants.KEY_2).get(Constants.KEY_3) == null;",
+            "   }",
+            "   boolean withCast(Map<String, Object> topLevelMap){",
+            "     return topLevelMap.get(Constants.KEY_1) == null ",
+            "       || ((Map<String,Object>) topLevelMap.get(Constants.KEY_1)).get(Constants.KEY_2) == null",
+            "       || ((Map<String,Object>) ",
+            "              ((Map<String,Object>) topLevelMap.get(Constants.KEY_1)).get(Constants.KEY_2))",
+            "                .get(Constants.KEY_3) == null;",
+            "   }",
```

Comment:
The above code reports:

```
/Test.java:14: warning: [NullAway] dereferenced expression ((Map<String,Object>) 
                .get(Constants.KEY_3) == null;
                ^
                ((Map<String,Object>) topLevelMap.get(Constants.KEY_1)).get(Constants.KEY_2)) is @Nullable
    (see http://t.uber.com/nullaway )
```

Reply:
Yes, this looks like a bug.  We should ideally be consistently stripping out casts when checking nullability of expressions.  I guess here the issue is maybe that `((Map<String,Object>) ((Map<String,Object>) topLevelMap.get(Constants.KEY_1)).get(Constants.KEY_2))` has a nested cast that isn't getting stripped out?

Reply:
Definitely the issue is related to nested casts, and, yeah, I imagine it is somewhere in our code for generating map APs. Only a few things are valid constants there, and it probably doesn't include `get((Foo) f)`, even when `get(f)` is a valid map AP. Will look into this soon, but just wanted to have it up as reminder while I finish something else :)

Reply:
But, you are right, we probably need only make a change to `AccessPath.getAccessPathForNodeWithMapGet(...)` [here](https://github.com/uber/NullAway/blob/master/nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPath.java#L328) to strip casts: